projects/common/src/user-preference/user-preference.service.ts

```diff
@@ -0,0 +1,66 @@
+import { HttpClient, HttpHeaders, HttpParams } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs';
+import { DynamicConfigurationService } from '../dynamic-configuration/dynamic-configuration.service';
+
+export interface IRequestOptions {
+  headers?: HttpHeaders;
+  observe?: 'body';
+  params?: HttpParams;
+  reportProgress?: boolean;
+  responseType?: 'json';
+  withCredentials?: boolean;
+  body?: object;
+}
+
+@Injectable({
+  providedIn: 'root'
+})
+export class UserPreferenceService {
+  public BASE_URL: string;
+  private static readonly BASE_URL_CONFIG_KEY: string = 'user_preference';
+  public constructor(
+    private readonly http: HttpClient,
+    private readonly dynamicConfigurationService: DynamicConfigurationService
+  ) {
+    this.BASE_URL = this.dynamicConfigurationService.getValueForUrlConfig(
+      UserPreferenceService.BASE_URL_CONFIG_KEY
+    ) as string;
+  }
+  private addUserEmailHeader(headers?: HttpHeaders): HttpHeaders {
+    if (headers) {
+      // TODO Read email form email service later
+      return headers.append('user.email', 'ht-user@razorpay.com');
+    }
+    const requestHeaders = new HttpHeaders();
+
+    return requestHeaders.append('user.email', 'ht-user@razorpay.com');
+  }
+  public get<T>(endPoint: string, options?: IRequestOptions): Observable<T> {
+    const requestOptions = { ...options };
```

Comment:
Have changed this to `const requestOptions = options || {}`
Need to create a new object because options can be undefined and hence if we do `options.headers = addHeader(options.headers)` , typescript complains that the object is possibly undefined. 
cannot do `options = options || {}` because of `no-parameter-reassignment` rule

Reply:
Options **will** be undefined for every use case we have at hand right now. I'd recommend removing it entirely, we don't need to hack around to make something work we're not even using.